Fixed issue with leaking GW configs

breakreturn opened the pull request:
The problem with GW configs leaking between caps
when using multiple containers is fixed. Relevant
service test is enabled again. Some additional
debug outpuut added as well.

Signed-off-by: Joakim Gross <joakim.gross@pelagicore.com>

DunderRoffe reviewed (changes requested):
Looks good but would be nice to keep debug stuff and comments as separate commits

frznlogic commented on common/gatewayconfig.cpp:
```diff
@@ -77,8 +88,8 @@ bool GatewayConfiguration::append(const std::string &id, json_t *sourceArray)
-        m_configMap[id] = sourceArray;
-        json_incref(sourceArray);
+        m_configMap[id] = json_deep_copy(sourceArray);
+        // we do not need to decref the sourceArray since we did a deep copy
```
Is this comment is needed for something?

breakreturn replied:
It's a comment for why we don't do anything with the source that we got from the caller. The reader be all like "hey, why no code here?", and we be like "cause we copy deep".

kursatkobya commented on agent/src/capability/baseconfigstore.cpp:
```diff
@@ -142,7 +143,7 @@ void BaseConfigStore::parseCapabilities(json_t *capabilities)
         }
 
-        log_debug() << "Found capability \"" << capName << "\", parsing gateways...";
+        log_debug() << "Found capability \"" << capName << "";
```
I guess you missed one " at the end
`log_debug() << "Found capability \"" << capName << "\"";`

breakreturn replied:
Yeah, will fix.

breakreturn committed:
Fixed issue with leaking GW config

The problem with GW configs leaking between caps
when using multiple containers is fixed. Relevant
service test is enabled again.

Signed-off-by: Joakim Gross <joakim.gross@pelagicore.com>

breakreturn committed:
Added comments and debug output related to caps

Signed-off-by: Joakim Gross <joakim.gross@pelagicore.com>

breakreturn commented:
@DunderRoffe I've split up the changes into two commits.
